packages/plugins/src/foundry/types.ts

```diff
@@ -74,3 +76,27 @@ export type FoundryToml = {
     }
   }
 }
+
+export type BuildInfoCache = {
+  _format: 'sphinx-build-info-cache-1'
+  entries: Record<string, BuildInfoCacheEntry>
+}
+
+/**
+ * @field contracts An array where each element corresponds to a contract in the
+ * `BuildInfo.output.contracts` object. We use this array to match collected contract init code and
+ * runtime bytecode to the corresponding artifact.
+ */
+export type BuildInfoCacheEntry = {
+  name: string
+  time: number
+  contracts: Array<{
+    fullyQualifiedName: string
+    bytecode: string
+    deployedBytecode: string
```

Comment:
We're adding a lot of information to the cache file now (this PR should roughly double its size). It's worth noting that if we increase the size of this file too much, we may encounter issues similar to those we encountered reading extremely large build-info files. 

If you don't remember, we are limited in the size of JSON files we can read using standard JSON.parse() by the maximum string length in NodeJS, which is ~512mb. If our cache exceeds this size, we would have to implement logic to use `stream-json` to read this file.

We don't need to address this issue right now since the cache would only exceed that limit in pretty extreme situations. I am just calling this out as a potential issue with adding more data to the cache file.

Reply:
There's already a [ticket](https://linear.app/chugsplash/issue/CHU-392/reduce-the-size-of-the-sphinx-cachejson-file) for that in the backlog, which I just put back in Triage in case we want to handle this. I'm not sure how big the cache file can get during standard usage, so I think it's reasonable to investigate sometime soon